mgmtfn/k8splugin/cniserver.go

```diff
@@ -0,0 +1,162 @@
+/***
+Copyright 2016 Cisco Systems Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8splugin
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/contiv/netplugin/mgmtfn/cniapi"
+	"github.com/contiv/netplugin/netplugin/plugin"
+	"github.com/gorilla/mux"
+)
+
+// contivKubeCfgFile holds credentials to access k8s api server
+const (
+	contivKubeCfgFile = "/opt/contiv/config/contiv.json"
```

Comment:
Well, kube set up uses these certs. Therefore, I don't consider it a security problem specific to contiv. There is also a different way of authenticating that I can look into, separately. However, I don't know that it is any more secure.